config/load.go

```diff
@@ -0,0 +1,298 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"github.com/adrg/xdg"
+	"github.com/mitchellh/go-homedir"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+func Load(cfg Config, cmd *cobra.Command, configurations ...any) error {
+	// allow for nested options to be specified via environment variables
+	// e.g. pod.context = APPNAME_POD_CONTEXT
+	v := viper.NewWithOptions(viper.EnvKeyReplacer(strings.NewReplacer(".", "_", "-", "_")))
+
+	return load(cfg, v, cmd, configurations...)
+}
+
+func LoadAt(cfg Config, cmd *cobra.Command, path string, configuration any) error {
+	t := reflect.TypeOf(configuration)
+	config := reflect.StructOf([]reflect.StructField{{
+		Name: upperFirst(path),
+		Type: t,
+		Tag:  reflect.StructTag(fmt.Sprintf(`json:"%s" yaml:"%s" mapstructure:"%s"`, path, path, path)),
+	}})
+
+	value := reflect.New(config)
+	value.Elem().Field(0).Set(reflect.ValueOf(configuration))
+
+	return Load(cfg, cmd, value.Interface())
+}
+
+func upperFirst(p string) string {
+	if len(p) < 2 {
+		return strings.ToUpper(p)
+	}
+	return strings.ToUpper(p[0:1]) + p[1:]
+}
+
+func load(cfg Config, v *viper.Viper, cmd *cobra.Command, configurations ...any) error {
```

Comment:
since the logger is being injected there should be a nil check around the top here and add a new discard logger if it is nil.